rotkehlchen/data/import_utils.py

```diff
@@ -0,0 +1,283 @@
+    available_operations: List[str]
+
+    @classmethod
+    def is_entry(cls, requested_operation: str) -> bool:
+        return requested_operation in cls.available_operations
+
+    @classmethod
+    def process_entry(cls, timestamp: Timestamp, data: BinanceCsvRow) -> None:
+        raise NotImplementedError('Should be implemented by subclass')
+
+
+class BinanceMultipleEntry(BinanceEntry):
+    @classmethod
+    def are_entries(cls, requested_operations: List) -> bool:
+        raise NotImplementedError('Should be implemented by subclass')
+
+    @classmethod
+    def process_entries(cls, timestamp: Timestamp, data: List[BinanceCsvRow]) -> int:
+        raise NotImplementedError('Should be implemented by subclass')
+
+
+class BinanceTradeEntry(BinanceMultipleEntry):
+    @classmethod
+    def are_entries(cls, requested_operations: List) -> bool:
+        counted = Counter(requested_operations)
+        fee = counted.pop('Fee', None)
+        keys = set(counted.keys())
+        return (
+            (
+                keys == {'Buy', 'Sell'} and counted['Buy'] % 2 == 0 and counted['Sell'] % 2 == 0 and  # noqa: E501
+                (not fee or counted['Sell'] + counted['Buy'] == fee * 2)
+            ) or
+            (keys == {'Buy'} and counted['Buy'] % 2 == 0 and (not fee or counted['Buy'] == fee * 2)) or  # noqa: E501
+            (keys == {'Sell'} and counted['Sell'] % 2 == 0 and (not fee or counted['Sell'] == fee * 2)) or  # noqa: E501
+            (keys == {'Transaction Related'} and counted['Transaction Related'] % 2 == 0) or
+            (keys == {'Small assets exchange BNB'} and counted['Small assets exchange BNB'] % 2 == 0)  # noqa: E501
+        )
+
+    @classmethod
+    def process_trades(cls, timestamp: Timestamp, data: List[BinanceCsvRow]) -> List[Trade]:
+        same_assets = True
+        assets: Dict[str, Optional[Asset]] = defaultdict(lambda: None)
+        for row in data:
+            if row['Operation'] == 'Fee':
+                cur_operation = 'Fee'
+            elif row['Change'] < 0:
+                cur_operation = 'Sold'
+            else:
+                cur_operation = 'Bought'
+            assets[cur_operation] = assets[cur_operation] or row['Coin']
+            if assets[cur_operation] != row['Coin']:
+                same_assets = False
+                break
+
+        if not same_assets:
+            for row in data:
+                try:
+                    price = PriceHistorian.query_historical_price(
+                        from_asset=row['Coin'],
+                        to_asset=A_USD,
+                        timestamp=timestamp,
+                    )
+                except NoPriceForGivenTimestamp:
+                    log.warning(f'Couldn\'t find price of {row["Coin"]} on {timestamp}')
+                    return []
+                row['usd_value'] = row['Change'] * price
+
+        def ops_key(x: BinanceCsvRow) -> bool:
+            return x['Operation'] == 'Fee'
```

Comment:
We try to avoid nested functions. Could this be defined as an auxiliar function outside the class? Maybe like `operation_is_fee` or similar

Reply:
Yep, changed!